htlcswitch/quiescer.go

```diff
@@ -403,3 +408,42 @@ func (q *Quiescer) initStfu(req StfuReq) {
 	q.localInit = true
 	q.activeQuiescenceReq = fn.Some(req)
 }
+
+// OnResume accepts a no return closure that will run when the quiescer is
+// resumed.
+func (q *Quiescer) OnResume(hook func()) {
+	q.Lock()
+	defer q.Unlock()
+
+	q.onResume(hook)
+}
+
+// onResume accepts a no return closure that will run when the quiescer is
+// resumed.
+func (q *Quiescer) onResume(hook func()) {
+	q.resumeQueue = append(q.resumeQueue, hook)
+}
+
+// Resume runs all of the deferred actions that have accumulated while the
+// channel has been quiescent and then resets the quiescer state to its initial
+// state.
+func (q *Quiescer) Resume() {
+	q.Lock()
+	defer q.Unlock()
+
+	q.resume()
+}
+
+// resume runs all of the deferred actions that have accumulated while the
+// channel has been quiescent and then resets the quiescer state to its initial
+// state.
+func (q *Quiescer) resume() {
```

Comment:
add some debug log here? think it'd be helpful for future development.

Reply:
Lmk if there are more sites that could benefit from this. I have bad intuition for where we want logging.